src/main/java/com/btkelly/gnag/reporters/BaseReporter.java

```diff
@@ -95,8 +99,22 @@ private T getReport(Project project) throws JAXBException {
         }
     }
 
-    private java.io.File getReportFile(Project project) {
-        return new java.io.File(project.getProjectDir(), getReportFilePath());
+    @Nullable
+    private java.io.File getReportFile(Project project) throws FileNotFoundException {
+        final File searchDirectory = new File(project.getProjectDir(), getReportDirectory());
+
+        final java.io.File[] matchingFiles = searchDirectory.listFiles(getReportFilenameFilter());
+
+        if (matchingFiles.length == 0) {
+            throw new FileNotFoundException("Could not locate any report file using filter "
+                + getReportFilenameFilter().toString());
+        } else if (matchingFiles.length >= 2) {
+            throw new FileNotFoundException("Could not locate unique report file using filter "
```

Comment:
In case we didn't clean and therefore have multiple (possibly-conflicting) lint reports for different variants.